libcperciva/network/network_connect.c

```diff
@@ -182,6 +188,24 @@ network_connect(struct sock_addr * const * sas,
 	return (network_connect_timeo(sas, NULL, callback, cookie));
 }
 
+/**
+ * network_connect_bind(sas, sa_b, callback, cookie):
+ * Iterate through the addresses in ${sas}, attempting to create and connect
+ * a non-blocking socket after binding to ${sa_b}.  Once connected, invoke
+ * ${callback}(${cookie}, s) where s is the connected socket; upon fatal error
+ * or if there are no addresses remaining to attempt, invoke
+ * ${callback}(${cookie}, -1).  Return  a cookie which can be passed to
+ * network_connect_cancel() in order to cancel the connection attempt.
+ */
+void *
+network_connect_bind(struct sock_addr * const * sas, struct sock_addr const * sa_b,
+    int (* callback)(void *, int), void * cookie)
+{
+
+	/* Let network_connect_timeo handle this. */
+	return (network_connect_timeo(sas, NULL, callback, cookie));
```

Comment:
Looks like you're not actually using `sa_b` here?

Reply:
Fixed.